### ISSUE TITLE
39 vulnerabilities found in`npm audit`

### ISSUE THREAD

**Issue by manojLondhe** [Sunbird-Ed/SunbirdEd-mobile#12]
Hi,

I was trying out cloning this repo and doing install locally, npm install indicated 39 vulnerabilities found.

Reporting issue - so this does not fall out of track.

```
added 1323 packages from 1287 contributors in 326.611s
[!] 39 vulnerabilities found [8116 packages audited]
    Severity: 5 Low | 30 Moderate | 4 High
    Run `npm audit` for more detail
```

npm audit report

```
my-lappy@manoj:~/GIT/sunbird-mobile$ npm audit
                                                                                
                       === npm audit security report ===                        
                                                                                
# Run  npm install --dev gulp@4.0.0  to resolve 4 vulnerabilities
SEMVER WARNING: Recommended action is a potentially breaking change
                                                                                
  High            Regular Expression Denial of Service                          
                                                                                
  Package         minimatch                                                     
                                                                                
  Dependency of   gulp [dev]                                                    
                                                                                
  Path            gulp > vinyl-fs > glob-stream > minimatch                     
                                                                                
  More info       https://nodesecurity.io/advisories/118                        
                                                                                


                                                                                
  High            Regular Expression Denial of Service                          
                                                                                
  Package         minimatch                                                     
                                                                                
  Dependency of   gulp [dev]                                                    
                                                                                
  Path            gulp > vinyl-fs > glob-watcher > gaze > globule > glob >      
                  minimatch                                                     
                                                                                
  More info       https://nodesecurity.io/advisories/118                        
                                                                                


                                                                                
  High            Regular Expression Denial of Service                          
                                                                                
  Package         minimatch                                                     
                                                                                
  Dependency of   gulp [dev]                                                    
                                                                                
  Path            gulp > vinyl-fs > glob-watcher > gaze > globule > minimatch   
                                                                                
  More info       https://nodesecurity.io/advisories/118                        
                                                                                


                                                                                
  Low             Prototype Pollution                                           
                                                                                
  Package         lodash                                                        
                                                                                
  Dependency of   gulp [dev]                                                    
                                                                                
  Path            gulp > vinyl-fs > glob-watcher > gaze > globule > lodash      
                                                                                
  More info       https://nodesecurity.io/advisories/577                        
                                                                                


# Run  npm update xmlbuilder --depth 5  to resolve 1 vulnerability
                                                                                
  Low             Prototype Pollution                                           
                                                                                
  Package         lodash                                                        
                                                                                
  Dependency of   protractor                                                    
                                                                                
  Path            protractor > webdriver-js-extender > selenium-webdriver >     
                  xml2js > xmlbuilder > lodash                                  
                                                                                
  More info       https://nodesecurity.io/advisories/577                        
                                                                                


                                                                                
                                 Manual Review                                  
             Some vulnerabilities require your attention to resolve             
                                                                                
          Visit https://go.npm.me/audit-guide for additional guidance           
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > node-gyp > request > hawk > boom > hoek           
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > node-gyp > request > hawk > cryptiles > boom >    
                  hoek                                                          
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > node-gyp > request > hawk > hoek                  
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > node-gyp > request > hawk > sntp > hoek           
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > request > hawk > boom > hoek                      
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > request > hawk > cryptiles > boom > hoek          
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > request > hawk > hoek                             
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > request > hawk > sntp > hoek                      
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > hipchat-notifier > request > hawk > boom >   
                  hoek                                                          
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > hipchat-notifier > request > hawk >          
                  cryptiles > boom > hoek                                       
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > hipchat-notifier > request > hawk > hoek     
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > hipchat-notifier > request > hawk > sntp >   
                  hoek                                                          
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > loggly > request > hawk > boom > hoek        
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > loggly > request > hawk > cryptiles > boom   
                  > hoek                                                        
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > loggly > request > hawk > hoek               
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > loggly > request > hawk > sntp > hoek        
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > slack-node > requestretry > request > hawk   
                  > boom > hoek                                                 
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > slack-node > requestretry > request > hawk   
                  > cryptiles > boom > hoek                                     
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > slack-node > requestretry > request > hawk   
                  > hoek                                                        
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > slack-node > requestretry > request > hawk   
                  > sntp > hoek                                                 
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   protractor                                                    
                                                                                
  Path            protractor > webdriver-manager > request > hawk > boom >      
                  hoek                                                          
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   protractor                                                    
                                                                                
  Path            protractor > webdriver-manager > request > hawk > cryptiles   
                  > boom > hoek                                                 
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   protractor                                                    
                                                                                
  Path            protractor > webdriver-manager > request > hawk > hoek        
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Prototype pollution                                           
                                                                                
  Package         hoek                                                          
                                                                                
  Patched in      > 4.2.0 < 5.0.0 || >= 5.0.3                                   
                                                                                
  Dependency of   protractor                                                    
                                                                                
  Path            protractor > webdriver-manager > request > hawk > sntp >      
                  hoek                                                          
                                                                                
  More info       https://nodesecurity.io/advisories/566                        
                                                                                
                                                                                
  Moderate        Memory Exposure                                               
                                                                                
  Package         tunnel-agent                                                  
                                                                                
  Patched in      >=0.6.0                                                       
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > node-gyp > request > tunnel-agent                 
                                                                                
  More info       https://nodesecurity.io/advisories/598                        
                                                                                
                                                                                
  Moderate        Memory Exposure                                               
                                                                                
  Package         tunnel-agent                                                  
                                                                                
  Patched in      >=0.6.0                                                       
                                                                                
  Dependency of   node-sass                                                     
                                                                                
  Path            node-sass > request > tunnel-agent                            
                                                                                
  More info       https://nodesecurity.io/advisories/598                        
                                                                                
                                                                                
  Moderate        Memory Exposure                                               
                                                                                
  Package         tunnel-agent                                                  
                                                                                
  Patched in      >=0.6.0                                                       
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > hipchat-notifier > request > tunnel-agent    
                                                                                
  More info       https://nodesecurity.io/advisories/598                        
                                                                                
                                                                                
  Moderate        Memory Exposure                                               
                                                                                
  Package         tunnel-agent                                                  
                                                                                
  Patched in      >=0.6.0                                                       
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > loggly > request > tunnel-agent              
                                                                                
  More info       https://nodesecurity.io/advisories/598                        
                                                                                
                                                                                
  Moderate        Memory Exposure                                               
                                                                                
  Package         tunnel-agent                                                  
                                                                                
  Patched in      >=0.6.0                                                       
                                                                                
  Dependency of   karma                                                         
                                                                                
  Path            karma > log4js > slack-node > requestretry > request >        
                  tunnel-agent                                                  
                                                                                
  More info       https://nodesecurity.io/advisories/598                        
                                                                                
                                                                                
  Moderate        Memory Exposure                                               
                                                                                
  Package         tunnel-agent                                                  
                                                                                
  Patched in      >=0.6.0                                                       
                                                                                
  Dependency of   protractor                                                    
                                                                                
  Path            protractor > webdriver-manager > request > tunnel-agent       
                                                                                
  More info       https://nodesecurity.io/advisories/598                        
                                                                                
                                                                                
  High            Regular Expression Denial of Service                          
                                                                                
  Package         minimatch                                                     
                                                                                
  Patched in      >=3.0.2                                                       
                                                                                
  Dependency of   gulp [dev]                                                    
                                                                                
  Path            gulp > vinyl-fs > glob-stream > glob > minimatch              
                                                                                
  More info       https://nodesecurity.io/advisories/118                        
                                                                                
                                                                                
  Low             Prototype Pollution                                           
                                                                                
  Package         lodash                                                        
                                                                                
  Patched in      >=4.17.5                                                      
                                                                                
  Dependency of   cordova-android                                               
                                                                                
  Path            cordova-android > cordova-common > plist > xmlbuilder >       
                  lodash                                                        
                                                                                
  More info       https://nodesecurity.io/advisories/577                        
                                                                                
                                                                                
  Low             Prototype Pollution                                           
                                                                                
  Package         lodash                                                        
                                                                                
  Patched in      >=4.17.5                                                      
                                                                                
  Dependency of   cordova-custom-config                                         
                                                                                
  Path            cordova-custom-config >                                       
                  cf59bc2dc96d4964892125f75b8aa3b070b52a99172f15453414344c57b…  
                  > xmlbuilder > lodash                                         
                                                                                
  More info       https://nodesecurity.io/advisories/577                        
                                                                                
                                                                                
  Low             Regular Expression Denial of Service                          
                                                                                
  Package         timespan                                                      
                                                                                
  Patched in      No patch available                                            
                                                                                
  Dependency of   karma [dev]                                                   
                                                                                
  Path            karma > log4js > loggly > timespan                            
                                                                                
  More info       https://nodesecurity.io/advisories/533                        
                                                                                

[!] 39 vulnerabilities found - Packages audited: 8116 (6910 dev, 893 optional)
    Severity: 5 Low | 30 Moderate | 4 High
```

**Comment by AmiableAnil** [comment]
@manojLondhe We have fixed this issue in release-1.11.5

**Comment by swayangjit** [comment]
@manojLondhe  All the fixes are pushed to master. So closing the issue.
